### PR TITLE
CSHARP-3142: mongocrypt.dll not present when using mongocsharpdriver in an 'SDK' project

### PR DESCRIPTION
**Pull request by MikalaiMazurenka**
https://jira.mongodb.org/browse/CSHARP-3142

Note: changes in .sln file were made because solution wasn't opening in visual studio 2019 and were made automatically.

### CODE REVIEW

File: bindings/cs/MongoDB.Libmongocrypt/MongoDB.Libmongocrypt.csproj [hunk 1]
```diff
@@ -73,13 +73,6 @@
     </Content>
   </ItemGroup>
 
-  <ItemGroup>
```
Comment (DmitryLukyanov) [inline bindings/cs/MongoDB.Libmongocrypt/MongoDB.Libmongocrypt.csproj:76]: it looks like the current solution requires nuget files in the global nuget storage to run local tests which is not correct. The local tests should be independent in this sense (as it was before). 
It's similar to say that to run local driver tests we need to add nuget files into global nuget storage firstly and only then be able to run unit tests. Please check it.

Comment (MikalaiMazurenka) [reply bindings/cs/MongoDB.Libmongocrypt/MongoDB.Libmongocrypt.csproj:76]: As we tested offline, tests are working fine (in test projects .dll is copied into output). But your comment made me recheck all the .dll paths and I figured out that by removing .targets I broke non SDK project restore, which I am fixing in the second commit.

**Review by MikalaiMazurenka — COMMENTED** [review]
Ready for review!

File: bindings/cs/MongoDB.Libmongocrypt/MongoDB.Libmongocrypt.csproj [hunk 2]
```diff
@@ -62,7 +62,7 @@
   <ItemGroup>
     <Content Include="$(LibMongoCryptBinaries)/mongocrypt.dll">
       <Pack>true</Pack>
-      <PackagePath>x64/native/windows</PackagePath>
+      <PackagePath>runtimes/win/native</PackagePath>
```
Comment (rstam) [inline bindings/cs/MongoDB.Libmongocrypt/MongoDB.Libmongocrypt.csproj:65]: Why the change?

File: bindings/cs/MongoDB.Libmongocrypt/MongoDB.Libmongocrypt.csproj [hunk 3]
```diff
@@ -74,7 +74,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <Content Include="MongoDB.Libmongocrypt.targets">
+    <Content Include="MongoDB.Libmongocrypt.props">
```
Comment (rstam) [inline bindings/cs/MongoDB.Libmongocrypt/MongoDB.Libmongocrypt.csproj:77]: I have no idea what this change means.

Comment (rstam) [reply bindings/cs/MongoDB.Libmongocrypt/MongoDB.Libmongocrypt.csproj:77]: Oh I see this is referencing a file YOU added. But why add this is a separate file?

File: bindings/cs/cs.sln [hunk 4]
```diff
@@ -3,13 +3,13 @@ Microsoft Visual Studio Solution File, Format Version 12.00
...
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "MongoDB.Libmongocrypt.Test", "MongoDB.Libmongocrypt.Test\MongoDB.Libmongocrypt.Test.csproj", "{F2D29C5D-DA83-456D-994F-57AB43EA9470}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "MongoDB.Libmongocrypt.Test", "MongoDB.Libmongocrypt.Test\MongoDB.Libmongocrypt.Test.csproj", "{F2D29C5D-DA83-456D-994F-57AB43EA9470}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "MongoDB.Libmongocrypt.Example", "MongoDB.Libmongocrypt.Example\MongoDB.Libmongocrypt.Example.csproj", "{536E0590-7287-4D73-A5D7-5B8EFD2945B8}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "MongoDB.Libmongocrypt.Example", "MongoDB.Libmongocrypt.Example\MongoDB.Libmongocrypt.Example.csproj", "{536E0590-7287-4D73-A5D7-5B8EFD2945B8}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "MongoDB.Libmongocrypt.Test32", "MongoDB.Libmongocrypt.Test32\MongoDB.Libmongocrypt.Test32.csproj", "{EBD0FAFF-4794-4346-9313-A286E278EDA7}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "MongoDB.Libmongocrypt.Test32", "MongoDB.Libmongocrypt.Test32\MongoDB.Libmongocrypt.Test32.csproj", "{EBD0FAFF-4794-4346-9313-A286E278EDA7}"
```
Comment (rstam) [inline bindings/cs/cs.sln:12]: I also have seen Visual Studio randomly rewrite these UUIDs when making other changes to the .sln file.

Usually they can be manually reverted to get rid of the diff.

File: bindings/cs/cs.sln [hunk 5]
```diff
@@ -20,46 +20,43 @@ Global
...
-		{536E0590-7287-4D73-A5D7-5B8EFD2945B8}.Debug|x64.Build.0 = Debug|x64
-		{536E0590-7287-4D73-A5D7-5B8EFD2945B8}.Debug|x86.ActiveCfg = Debug|x64
-		{536E0590-7287-4D73-A5D7-5B8EFD2945B8}.Debug|x86.Build.0 = Debug|x64
-		{536E0590-7287-4D73-A5D7-5B8EFD2945B8}.Release|Any CPU.ActiveCfg = Release|x64
-		{536E0590-7287-4D73-A5D7-5B8EFD2945B8}.Release|Any CPU.Build.0 = Release|x64
-		{536E0590-7287-4D73-A5D7-5B8EFD2945B8}.Release|x64.ActiveCfg = Release|x64
-		{536E0590-7287-4D73-A5D7-5B8EFD2945B8}.Release|x64.Build.0 = Release|x64
-		{536E0590-7287-4D73-A5D7-5B8EFD2945B8}.Release|x86.ActiveCfg = Release|x64
-		{536E0590-7287-4D73-A5D7-5B8EFD2945B8}.Release|x86.Build.0 = Release|x64
+		{90EA4DBB-69E6-426D-985A-C66D65A2E63A}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
```
Comment (rstam) [inline bindings/cs/cs.sln:24]: Why Any Cpu?

**Review by rstam — COMMENTED** [review]
I really have no idea what this pull request is doing. I've made a few comments but I am going to remove myself as a reviewer.

**Comment by rstam** [comment]
I was unable to remove myself as a reviewer. Please remove me.

**Review by DmitryLukyanov — COMMENTED** [review]
NOTE: I documented expected behavior for where native libraries should be placed: https://docs.google.com/document/d/1FYxq9LPqyCjD65URosiP9v3yruB4dWPmI64NwKLii0I/

**Review by DmitryLukyanov — CHANGES_REQUESTED** [review]
-

File: bindings/cs/MongoDB.Libmongocrypt/LibraryLoader.cs [hunk 6]
```diff
@@ -82,12 +81,12 @@ public LibraryLoader()
                 _loader = new LinuxLibrary(path);
             }
             else if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-
 #endif
             {
                 string[] suffixPaths = new[]
                 {
-                    @"..\..\x64\native\windows\",
+                    "../../runtimes/win/native/",
```
[the remark this answers is not in the document]
Comment (MikalaiMazurenka) [reply bindings/cs/MongoDB.Libmongocrypt/LibraryLoader.cs]: Done.

File: bindings/cs/MongoDB.Libmongocrypt/LibraryLoader.cs [hunk 7]
```diff
@@ -107,7 +106,6 @@ private string FindLibrary(IList<string> basePaths, string[] suffixPaths, string
                     if (File.Exists(path))
                     {
                         // TODO - .NET Standard 2.0
-                        //Trace.WriteLine("Load path: " + path);
```
[the remark this answers is not in the document]
Comment (MikalaiMazurenka) [reply bindings/cs/MongoDB.Libmongocrypt/LibraryLoader.cs]: Agreed.
Done.

File: bindings/cs/MongoDB.Libmongocrypt/MongoDB.Libmongocrypt.csproj (hunk 2 above)
Comment (MikalaiMazurenka) [reply bindings/cs/MongoDB.Libmongocrypt/MongoDB.Libmongocrypt.csproj:65]: Yes, this is the magic folder.

File: bindings/cs/MongoDB.Libmongocrypt/MongoDB.Libmongocrypt.csproj (hunk 3 above)
Comment (MikalaiMazurenka) [reply bindings/cs/MongoDB.Libmongocrypt/MongoDB.Libmongocrypt.csproj]: Done.

File: bindings/cs/cs.sln (hunk 4 above)
Comment (MikalaiMazurenka) [reply bindings/cs/cs.sln]: Done.

File: bindings/cs/cs.sln (hunk 5 above)
Comment (MikalaiMazurenka) [reply bindings/cs/cs.sln]: This is the change which allowed vs2019 to open solution without issues, but I agree, that we should only build x64 version, so I'm reverting this.
Done.

**Review by MikalaiMazurenka — COMMENTED** [review]
Ready for review!

**Comment by rstam** [comment]
I removed myself as a reviewer. I think we need reviewers who have ownership responsibility for mongocrypt.dll.

File: bindings/cs/MongoDB.Libmongocrypt/MongoDB.Libmongocrypt.targets [hunk 8]
```diff
@@ -1,10 +1,9 @@
-<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-
-    <ItemGroup Condition="$([MSBuild]::IsOsPlatform('Windows'))">
-        <Content Include="$(MSBuildThisFileDirectory)../x64/native/windows/mongocrypt.dll">
-            <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
-            <Link>mongocrypt.dll</Link>
-        </Content>
-    </ItemGroup>
+<Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (DmitryLukyanov) [inline bindings/cs/MongoDB.Libmongocrypt/MongoDB.Libmongocrypt.targets:1]: Not sure about this change. I remember that we definitely had a case where we needed exactly the previous structure. But I wasn't able to recall it for now. 
So, I will approve it

**Review by DmitryLukyanov — APPROVED** [review]
LGTM
